### PR TITLE
Update foundation.d.ts

### PR DESCRIPTION
**Pull request by stcp**
#9593 Updated some options.
Added ResponsiveMenu and ResponsiveToggle types.
Added defaults to all plugins.

### CODE REVIEW

**Comment by IamManchanda** [comment]
i hate this travis ci .... i must say ..... network connectivity issues always

File: dist/js/foundation.d.ts [hunk 1]
```diff
@@ -62,7 +55,8 @@ declare module FoundationSites {
     formErrorSelector?: string;
     formErrorClass?: string;
     liveValidate?: boolean;
-    validators?: any;
+    validators?: {[key:string]:Function};
+    patterns?:{[key:string]:RegExp};
```
Comment (DaSchTour) [inline dist/js/foundation.d.ts]: I'm not sure if this will provide any already present pattern.

File: dist/js/foundation.d.ts [hunk 2]
```diff
@@ -73,6 +67,10 @@ declare module FoundationSites {
     destroy(): void;
   }
 
+  interface AccordionDefaults extends IAccordionOptions{
```
Comment (DaSchTour) [inline dist/js/foundation.d.ts]: Empty interface? So why not reference IAccordionOptions directly. From my look on the JS I'm also not sure if Defaults and options are the same. From what I see they are not.

File: dist/js/foundation.d.ts [hunk 3]
```diff
@@ -88,6 +86,10 @@ declare module FoundationSites {
     destroy(): void;
   }
 
+  interface AccordionMenuDefaults extends IAccordionMenuOptions{
```
Comment (DaSchTour) [inline dist/js/foundation.d.ts]: Empty interfaces?

File: dist/js/foundation.d.ts [hunk 4]
```diff
@@ -333,21 +438,28 @@ declare module FoundationSites {
...
+
+  }
+
   interface ITooltipOptions {
+    disableForTouch?: boolean;
     hoverDelay?: number;
     fadeInDuration?: number;
     fadeOutDuration?: number;
-    disableHover?: boolean;
+    disableHover?: number;
```
Comment (DaSchTour) [inline dist/js/foundation.d.ts]: Documentation says this is a boolean. http://foundation.zurb.com/sites/docs/tooltip.html

File: dist/js/foundation.d.ts [hunk 5]
```diff
@@ -24,16 +25,8 @@ declare module FoundationSites {
     destroy(): void;
   }
 
-  interface AbideDefaults {
```
Comment (DaSchTour) [inline dist/js/foundation.d.ts]: Why this is removed? This are the default properties.

**Review by DaSchTour — CHANGES_REQUESTED** [review]
Why are there so many empty interfaces?

**Comment by kball** [comment]
@DaSchTour @stcp what is the status of this PR?

**Comment by DaSchTour** [comment]
@kball I think we shouldn't change anything here. We should do the work to shift to TypeScript and get full typings from there.

**Comment by kball** [comment]
@DaSchTour Given we'll likely do at least one more release (6.4) prior to any sort of TypeScript shift, are our current typings good enough? We're likely looking at July for 6.5

**Comment by DaSchTour** [comment]
@kball I work with them regularly and they serve well. I'm note sure about the changes made here and the consequences they may have and the author didn't answer any of my questions :/

**Comment by kball** [comment]
Seems reasonable. Closing in lieu of any additional responses indicating why we'd need these.

**Pull request closed without merging**